aten/src/ATen/cpu/vec/vec256/vec256_complex_double.h

```diff
@@ -121,7 +121,20 @@ template <> class Vectorized<c10::complex<double>> {
     return _mm256_hadd_pd(val_2, val_2);            // a*a+b*b a*a+b*b
   }
   __m256d abs_() const {
-    return _mm256_sqrt_pd(abs_2_());                // abs     abs
+    // values: a + ib
+    // not using abs_2_ to prevent overflow/underflow for large/small numbers
+    auto mask = _mm256_set1_pd(-0.f);
+    auto fabs_val = _mm256_andnot_pd(mask, values);     // |a|    |b|
+    auto fabs_shf = _mm256_permute_pd(fabs_val, 0x05);  // |b|    |a|
+    auto fabs_max = _mm256_max_pd(fabs_val, fabs_shf);  // max    max
+    auto fabs_min = _mm256_min_pd(fabs_val, fabs_shf);  // min    min
+    // following: max * sqrt(1 + min / max)
```

Comment:
Actually, being reminded of `hypot`'s implementation is making me think. Say we are computing `x / y`, why don't we scale the components of `x` and `y` by `1 / max(abs(y.real), abs(y.imag))`. `y_scaled.abs_2_()` will be in the vicinity of 1, so we avoid any over/underflow issues and we also avoid the expensive `sqrt` computation.

Reply:
We could even use `rcp_ps` to compute the scale factor, since it cancels out so we don't care about precision.

Reply:
That's a good idea! Let me try benchmarking it

Reply:
A neat preconditioner for the simplest linear system in the world :P

Reply:
Who would've thought that complex division is actually complex. :P
@peterbell10: I did benchmark it directly on a standalone c++ file and your idea is the fastest. However, it is about 50% slower than the original implementation. My implementation (that uses numerically-stabilized abs) was about 3x slower though.
Here's the code I used for benchmarking: https://gist.github.com/mfkasim1/14190cc7e701d605767b85a33772bd44
The results are:
```
time for orig div: 8058613
0.0000003767013550
time for new  div: 23312408
0.0000001001358032
time for new div2: 13650801
0.0000001001358032
time for div_map : 54636386
0.0000000190734863
```